Add boolean flag to control MC true branch address setup

forhad-h2030 opened the pull request:
I have added support for MC true track variables, controlled by the boolean flag `is_mc`.  
This flag can be enabled using the function `set_mc(true);`.   Actually I forgot to add this in the last pull request.

When `is_mc` is enabled, the file manager expects the input RUS file to contain the true track variables (`gCharge`, `gTrackID`, `gvx`, `gvy`, `gvz`, `gpx`, `gpy`, `gpz`, etc.).  
If `is_mc` is disabled, these branches are skipped, allowing the file manager to run normally with real data files that do not include MC truth information.

Relevant code block: https://github.com/E1039-Collaboration/e1039-core/blob/devel_RUS/packages/RUS/Fun4AllRUSInputManager.cc#L233-L256

forhad-h2030 committed:
adding a required bool for mc true branches to set the branch address

liuk commented:
Do you really need this additional data filed? Why not something like `if (_tin->GetBranch(gCharge) != nullptr) { //... set the branches }`"

forhad-h2030 commented:
> Do you really need this additional data filed? Why not something like `if (_tin->GetBranch(gCharge) != nullptr) { //... set the branches }`"

Thanks Kun. This is better and very automatic. I will update it accordingly.

forhad-h2030 committed:
Removed redundant functions related to manual MC mode control

forhad-h2030 commented:
@liuk @knakano0524: 

Deleted `SetMCTrueMode()` from `Fun4AllRUSOutputManager`.
Dropped `set_mc()` and `get_mc()` from `Fun4AllRUSInputManager`.
MC mode is now detected automatically based on ROOT branches or node presence.

ref  major changes:  

1. https://github.com/E1039-Collaboration/e1039-core/blob/devel_RUS/packages/RUS/Fun4AllRUSInputManager.cc#L232 
2. https://github.com/E1039-Collaboration/e1039-core/blob/devel_RUS/packages/RUS/Fun4AllRUSOutputManager.cc#L103

knakano0524 commented:
The SQTruthTrackVector node should not be created in case of !is_mc.

forhad-h2030 committed:
Refactor Fun4AllRUSInputManager and write a separate MakeNode() function

forhad-h2030 commented:
> The SQTruthTrackVector node should not be created in case of !is_mc.

Thanks. As you suggested,  I have made the following changes: 

- Added a dedicated `MakeNode()` function to handle node creation cleanly.  https://github.com/E1039-Collaboration/e1039-core/blob/devel_RUS/packages/RUS/Fun4AllRUSInputManager.cc#L32 
-  Ensured `MakeNode()` is safely called only once in  `Fun4AllRUSInputManager::fileopen()`: https://github.com/E1039-Collaboration/e1039-core/blob/devel_RUS/packages/RUS/Fun4AllRUSInputManager.cc#L441
-  The `SQTruthTrackVector` will not be created in case of `!is_mc`. https://github.com/E1039-Collaboration/e1039-core/blob/devel_RUS/packages/RUS/Fun4AllRUSInputManager.cc#L469

I have rebuild the e1039-core, and checked RUS I/O for consistency. All worked smoothly.

knakano0524 commented:
The updated code looks fine.  You can merge this PR.
